diff --git a/trunk/user/www/dict/CN.dict b/trunk/user/www/dict/CN.dict
index 28f7dd5bd..63fd44170 100644
--- a/trunk/user/www/dict/CN.dict
+++ b/trunk/user/www/dict/CN.dict
@@ -804,8 +804,8 @@ menu5_14_2=监听地址:
 menu5_14_3=监听端口:
 menu5_14_4=上游DNS:
 menu5_14_5=启用双向过滤
-menu5_15=DNS-forwarder
-menu5_15_1=启用DNS-forwarder
+menu5_15=DNS转发
+menu5_15_1=启用DNS转发
 menu5_16=科学上网
 menu5_16_1=全局设置
 menu5_16_2=启用ShadowSocks
