diff --git a/trunk/user/www/dict/CN.dict b/trunk/user/www/dict/CN.dict
index 28f7dd5bd..7040fdf7d 100644
--- a/trunk/user/www/dict/CN.dict
+++ b/trunk/user/www/dict/CN.dict
@@ -804,8 +804,8 @@ menu5_14_2=监听地址:
 menu5_14_3=监听端口:
 menu5_14_4=上游DNS:
 menu5_14_5=启用双向过滤
-menu5_15=DNS-forwarder
-menu5_15_1=启用DNS-forwarder
+menu5_15=DNS服务
+menu5_15_1=DNS转发
 menu5_16=科学上网
 menu5_16_1=全局设置
 menu5_16_2=启用ShadowSocks
diff --git a/trunk/user/www/n56u_ribbon_fixed/dns-forwarder.asp b/trunk/user/www/n56u_ribbon_fixed/dns-forwarder.asp
index aa3052ed4..a17d4674d 100644
--- a/trunk/user/www/n56u_ribbon_fixed/dns-forwarder.asp
+++ b/trunk/user/www/n56u_ribbon_fixed/dns-forwarder.asp
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
 <head>
-<title><#Web_Title#> - <#menu5_15#></title>
+<title><#Web_Title#> - <#menu5_15_1#></title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 <meta http-equiv="Pragma" content="no-cache">
 <meta http-equiv="Expires" content="-1">
@@ -117,7 +117,7 @@ function fill_status(status_code){
                 <div class="row-fluid">
                     <div class="span12">
                         <div class="box well grad_colour_dark_blue">
-                            <h2 class="box_head round_top"><#menu5_15#></h2>
+                            <h2 class="box_head round_top"><#menu5_15_1#></h2>
                             <div class="round_bottom">
                                 <div class="row-fluid">
                                     <div id="tabMenu" class="submenuBlock"></div>
@@ -126,7 +126,7 @@ function fill_status(status_code){
                                             <td id="dnsforwarder_status" colspan="3"></td>
                                         </tr>
 
-                                        <tr> <th><#menu5_15_1#></th>
+                                        <tr> <th>启用转发</th>
                                             <td>
                                                 <div class="main_itoggle">
                                                 <div id="dns_forwarder_enable_on_of">
diff --git a/trunk/user/www/n56u_ribbon_fixed/state.js b/trunk/user/www/n56u_ribbon_fixed/state.js
index 5fa1b3dfa..f77289b9a 100644
--- a/trunk/user/www/n56u_ribbon_fixed/state.js
+++ b/trunk/user/www/n56u_ribbon_fixed/state.js
@@ -395,7 +395,7 @@ if (found_app_scutclient()){
 	tabtitle[10] = new Array("", "<#menu5_1_1#>","<#menu5_13_log#>");
 }
 if (found_app_dnsforwarder()){
-	tabtitle[11] = new Array("", "<#menu5_1_1#>");
+	tabtitle[11] = new Array("", "<#menu5_15_1#>");
 }
 if (found_app_shadowsocks()){
 	tabtitle[12] = new Array("", "<#menu5_1_1#>","<#menu5_16_20#>");
