diff --git a/trunk/user/www/n56u_ribbon_fixed/Advanced_aliddns.asp b/trunk/user/www/n56u_ribbon_fixed/Advanced_aliddns.asp
index 1cf6237c2..86b3fd694 100644
--- a/trunk/user/www/n56u_ribbon_fixed/Advanced_aliddns.asp
+++ b/trunk/user/www/n56u_ribbon_fixed/Advanced_aliddns.asp
@@ -48,6 +48,7 @@ function initial(){
 
 function showmenu(){
 showhide_div('zelink', found_app_zerotier());
+showhide_div('frplink', found_app_frp());
 }
 
 function textarea_scripts_enabled(v){
@@ -138,16 +139,19 @@ function change_aliddns_enable_bridge(mflag){
 						<div class="box well grad_colour_dark_blue">
 							<h2 class="box_head round_top"><#menu5_23#> - <#menu5_30#></h2>
 							<div class="round_bottom">
-							<div>
-                            <ul class="nav nav-tabs" style="margin-bottom: 10px;">
-								<li class="active">
-                                    <a href="Advanced_aliddns.asp"><#menu5_23_1#></a>
-                                </li>
-								<li id="zelink" style="display:none">
-                                    <a href="Advanced_zerotier.asp"><#menu5_32_1#></a>
-                                </li>
-                            </ul>
-                        </div>
+								<div>
+									<ul class="nav nav-tabs" style="margin-bottom: 10px;">
+										<li class="active">
+											<a href="Advanced_aliddns.asp"><#menu5_23_1#></a>
+										</li>
+										<li id="zelink" style="display:none">
+											<a href="Advanced_zerotier.asp"><#menu5_32_1#></a>
+										</li>
+										<li id="frplink" style="display:none">
+											<a href="Advanced_frp.asp">FRP</a>
+										</li>
+									</ul>
+								</div>
 								<div class="row-fluid">
 									<div id="tabMenu" class="submenuBlock"></div>
 									<div class="alert alert-info" style="margin: 10px;">使用 Aliddns 实现顶级个人域名的 ddns 服务。 <a href="https://www.aliyun.com" target="blank"><i><u>Aliddns 主页</u></i></a>
diff --git a/trunk/user/www/n56u_ribbon_fixed/Advanced_frp.asp b/trunk/user/www/n56u_ribbon_fixed/Advanced_frp.asp
index 5c85a4b4c..9e3991e3e 100644
--- a/trunk/user/www/n56u_ribbon_fixed/Advanced_frp.asp
+++ b/trunk/user/www/n56u_ribbon_fixed/Advanced_frp.asp
@@ -1,7 +1,7 @@
 <!DOCTYPE html>
 <html>
 <head>
-<title><#Web_Title#> - 内网穿透</title>
+<title><#Web_Title#> - FRP内网穿透</title>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
 <meta http-equiv="Pragma" content="no-cache">
 <meta http-equiv="Expires" content="-1">
@@ -37,7 +37,8 @@ $j(document).ready(function() {
 
 function initial(){
 	show_banner(2);
-	show_menu(5,18);
+	show_menu(5,17,0);
+	showmenu();
 	show_footer();
 
 fill_status(frpc_status());
@@ -46,6 +47,11 @@ fill_status2(frps_status());
 		textarea_scripts_enabled(0);
 }
 
+function showmenu(){
+showhide_div('allink', found_app_aliddns());
+showhide_div('zelink', found_app_zerotier());
+}
+
 function textarea_scripts_enabled(v){
 	inputCtrl(document.form['scripts.frp_script.sh'], v);
 }
@@ -137,8 +143,21 @@ function done_validating(action){
 				<div class="row-fluid">
 					<div class="span12">
 						<div class="box well grad_colour_dark_blue">
-							<h2 class="box_head round_top">Frp - 内网穿透</h2>
+							<h2 class="box_head round_top">Frp - <#menu5_30#></h2>
 							<div class="round_bottom">
+								<div>
+									<ul class="nav nav-tabs" style="margin-bottom: 10px;">
+										<li id="allink" style="display:none">
+											<a href="Advanced_aliddns.asp"><#menu5_23_1#></a>
+										</li>
+										<li id="zelink" style="display:none">
+											<a href="Advanced_zerotier.asp"><#menu5_32_1#></a>
+										</li>
+										<li class="active">
+											<a href="Advanced_frp.asp">FRP</a>
+										</li>
+									</ul>
+								</div>
 								<div class="row-fluid">
 									<div id="tabMenu" class="submenuBlock"></div>
 									<div class="alert alert-info" style="margin: 10px;">frp 是一个可用于内网穿透的高性能的反向代理应用，支持 tcp, udp 协议，为 http 和 https 应用协议提供了额外的能力，且尝试性支持了点对点穿透。
diff --git a/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp b/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp
index 57f5c1047..76046ba28 100644
--- a/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp
+++ b/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp
@@ -56,6 +56,7 @@ showMRULESList();
 }
 function showmenu(){
 showhide_div('allink', found_app_aliddns());
+showhide_div('frplink', found_app_frp());
 }
 function applyRule(){
 //	if(validForm()){
@@ -188,16 +189,19 @@ function showMRULESList(){
 						<div class="box well grad_colour_dark_blue">
 							<h2 class="box_head round_top"><#menu5_32#> - <#menu5_30#></h2>
 							<div class="round_bottom">
-							<div>
-                            <ul class="nav nav-tabs" style="margin-bottom: 10px;">
-								<li id="allink" style="display:none">
-                                    <a href="Advanced_aliddns.asp"><#menu5_23_1#></a>
-                                </li>
-								<li class="active">
-                                    <a href="Advanced_zerotier.asp"><#menu5_32_1#></a>
-                                </li>
-                            </ul>
-                        </div>
+								<div>
+									<ul class="nav nav-tabs" style="margin-bottom: 10px;">
+										<li id="allink" style="display:none">
+											<a href="Advanced_aliddns.asp"><#menu5_23_1#></a>
+										</li>
+										<li class="active">
+											<a href="Advanced_zerotier.asp"><#menu5_32_1#></a>
+										</li>
+										<li id="frplink" style="display:none">
+											<a href="Advanced_frp.asp">FRP</a>
+										</li>
+									</ul>
+								</div>
 								<div class="row-fluid">
 									<div id="tabMenu" class="submenuBlock"></div>
 									<div class="alert alert-info" style="margin: 10px;">
diff --git a/trunk/user/www/n56u_ribbon_fixed/state.js b/trunk/user/www/n56u_ribbon_fixed/state.js
index 5fa1b3dfa..f25d96312 100644
--- a/trunk/user/www/n56u_ribbon_fixed/state.js
+++ b/trunk/user/www/n56u_ribbon_fixed/state.js
@@ -422,13 +422,10 @@ if (found_app_adguardhome()){
 }
 if (found_app_aliddns()){
 	tabtitle[16] = new Array("", "<#menu5_30#>");
-}else{
-if (found_app_zerotier()){
+}else if (found_app_zerotier()){
 	tabtitle[16] = new Array("", "<#menu5_32#>");
-}
-}
-if (found_app_frp()){
-	tabtitle[17] = new Array("", "<#menu5_25_1#>");
+}else if (found_app_frp()){
+	tabtitle[16] = new Array("", "<#menu5_25_1#>");
 }
 if (found_app_caddy()){
 	tabtitle[18] = new Array("", "<#menu5_27_1#>");
@@ -484,10 +481,9 @@ if (found_app_aliddns()){
 }else if (found_app_zerotier()){
 	zerotier_array = new Array("","Advanced_zerotier.asp");
 	tablink[16] = (zerotier_array);
-}
-if (found_app_frp()){
+}else if (found_app_frp()){
 	frp_array = new Array("","Advanced_frp.asp");
-	tablink[17] = (frp_array);
+	tablink[16] = (frp_array);
 }
 if (found_app_caddy()){
 	caddy_array = new Array("","Advanced_caddy.asp");
@@ -533,10 +529,8 @@ if (found_app_aliddns()){
 	menuL2_title.push("<#menu5_30#>");
 } else if (found_app_zerotier()){
 	menuL2_title.push("<#menu5_30#>");
-} else menuL2_title.push("");
-
-if (found_app_frp()){
-	menuL2_title.push("<#menu5_25#>");
+} else if (found_app_frp()){
+	menuL2_title.push("<#menu5_30#>");
 } else menuL2_title.push("");
 
 if (found_app_caddy()){
@@ -577,8 +571,7 @@ if (found_app_aliddns()){
 	menuL2_link.push(aliddns_array[1]);
 } else if (found_app_zerotier()){
 	menuL2_link.push(zerotier_array[1]);
-} else menuL2_link.push("");
-if (found_app_frp()){
+} else if (found_app_frp()){
 	menuL2_link.push(frp_array[1]);
 } else menuL2_link.push("");
 if (found_app_caddy()){
