diff --git a/trunk/user/httpd/variables.c b/trunk/user/httpd/variables.c
index 621f1886d..f46b0502a 100644
--- a/trunk/user/httpd/variables.c
+++ b/trunk/user/httpd/variables.c
@@ -108,8 +108,8 @@
 		};
 	struct variable variables_ZeroConf_ZeroList[] = {
 			{"zero_enable_x", "24", NULL, FALSE},
-			{"zero_ip_x", "24", NULL, FALSE},
-			{"zero_route_x", "24", NULL, FALSE},
+			{"zero_id_x", "24", NULL, FALSE},
+			{"zero_note_x", "24", NULL, FALSE},
 			{0,0,0,0}
 		};
 	struct variable variables_SspConf_SspList[] = {
@@ -983,8 +983,8 @@
 			{"zerotier_enable", "", NULL, EVM_RESTART_ZEROTIER},
 			{"zerotier_id", "", NULL, EVM_RESTART_ZEROTIER},
 			{"zerotier_nat", "", NULL, EVM_RESTART_ZEROTIER},
-			{"zero_staticnum_x", "", NULL, EVM_RESTART_ZEROTIER},
-			{"ZeroList", "Group", ARGV((char*)variables_ZeroConf_ZeroList, "8", "55", "zero_staticnum_x"), EVM_RESTART_ZEROTIER},
+			{"zero_netsnum_x", "", NULL, EVM_RESTART_ZEROTIER},
+			{"ZeroList", "Group", ARGV((char*)variables_ZeroConf_ZeroList, "8", "55", "zero_netsnum_x"), EVM_RESTART_ZEROTIER},
 			{0,0,0,0}
 	};
 #endif
diff --git a/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp b/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp
index 57f5c1047..bea537c1a 100644
--- a/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp
+++ b/trunk/user/www/n56u_ribbon_fixed/Advanced_zerotier.asp
@@ -76,28 +76,21 @@ function done_validating(action){
 function markGroupRULES(o, c, b) {
 	document.form.group_id.value = "ZeroList";
 	if(b == " Add "){
-		if (document.form.zero_staticnum_x_0.value >= c){
+		if (document.form.zero_netsnum_x_0.value >= c){
 			alert("<#JS_itemlimit1#> " + c + " <#JS_itemlimit2#>");
 			return false;
-		}else if (document.form.zero_ip_x_0.value==""){
+		}else if (document.form.zero_id_x_0.value==""){
 			alert("<#JS_fieldblank#>");
-			document.form.zero_ip_x_0.focus();
-			document.form.zero_ip_x_0.select();
-			return false;
-		}else if(document.form.zero_route_x_0.value==""){
-			alert("<#JS_fieldblank#>");
-			document.form.zero_route_x_0.focus();
-			document.form.zero_route_x_0.select();
+			document.form.zero_id_x_0.focus();
+			document.form.zero_id_x_0.select();
 			return false;
 		}else{
 			for(i=0; i<m_list.length; i++){
-				if(document.form.zero_ip_x_0.value==m_list[i][0]) {
-				if(document.form.zero_route_x_0.value==m_list[i][1]) {
-					alert('<#JS_duplicate#>' + ' (' + m_list[i][1] + ')' );
-					document.form.zero_ip_x_0.focus();
-					document.form.zero_ip_x_0.select();
+				if(document.form.zero_id_x_0.value==m_list[i][0]) {
+					alert('<#JS_duplicate#>' + ' (' + m_list[i][0] + ')' );
+					document.form.zero_id_x_0.focus();
+					document.form.zero_id_x_0.select();
 					return false;
-					}
 				}
 			}
 		}
@@ -164,7 +157,7 @@ function showMRULESList(){
 	<input type="hidden" name="group_id" value="ZeroList">
 	<input type="hidden" name="action_mode" value="">
 	<input type="hidden" name="action_script" value="">
-	<input type="hidden" name="zero_staticnum_x_0" value="<% nvram_get_x("ZeroList", "zero_staticnum_x"); %>" readonly="1" />
+	<input type="hidden" name="zero_netsnum_x_0" value="<% nvram_get_x("ZeroList", "zero_netsnum_x"); %>" readonly="1" />
 
 	<div class="container-fluid">
 		<div class="row-fluid">
@@ -201,14 +194,14 @@ function showMRULESList(){
 								<div class="row-fluid">
 									<div id="tabMenu" class="submenuBlock"></div>
 									<div class="alert alert-info" style="margin: 10px;">
-									<p>Zerotier是一个开源，跨平台，而且适合内网穿透互联的傻瓜配置虚拟 VPN LAN<br>
+									<p style="margin: 0;">Zerotier是一个开源，跨平台，而且适合内网穿透互联的傻瓜配置虚拟 VPN LAN<br>
 									</p>
 									</div>
 
 									<table width="100%" align="center" cellpadding="4" cellspacing="0" class="table">
 										<tr>
-											<th width="30%" style="border-top: 0 none;">启用ZeroTier</th>
-											<td style="border-top: 0 none;">
+											<th>启用ZeroTier</th>
+											<td>
 													<div class="main_itoggle">
 													<div id="zerotier_enable_on_of">
 														<input type="checkbox" id="zerotier_enable_fake" <% nvram_match_x("", "zerotier_enable", "1", "value=1 checked"); %><% nvram_match_x("", "zerotier_enable", "0", "value=0"); %>  />
@@ -221,15 +214,9 @@ function showMRULESList(){
 											</td>
 
 										</tr>
-										</tr>
-<tr><th>ZeroTier Network ID</th>
-				<td>
-					<input type="text" class="input" name="zerotier_id" id="zerotier_id" style="width: 200px" value="<% nvram_get_x("","zerotier_id"); %>" />
-				</td>
-			</tr>
-			<tr>
-											<th width="30%" style="border-top: 0 none;">自动允许客户端NAT</th>
-											<td style="border-top: 0 none;">
+										<tr>
+											<th>允许转发到LAN</th>
+											<td>
 													<div class="main_itoggle">
 													<div id="zerotier_nat_on_of">
 														<input type="checkbox" id="zerotier_nat_fake" <% nvram_match_x("", "zerotier_nat", "1", "value=1 checked"); %><% nvram_match_x("", "zerotier_nat", "0", "value=0"); %>  />
@@ -239,30 +226,28 @@ function showMRULESList(){
 													<input type="radio" value="1" name="zerotier_nat" id="zerotier_nat_1" class="input" value="1" <% nvram_match_x("", "zerotier_nat", "1", "checked"); %> /><#checkbox_Yes#>
 													<input type="radio" value="0" name="zerotier_nat" id="zerotier_nat_0" class="input" value="0" <% nvram_match_x("", "zerotier_nat", "0", "checked"); %> /><#checkbox_No#>
 												</div>
-												 允许Zerotier的拨入客户端访问路由器LAN资源（需要在 Zerotier管理页面设定到LAN网段的路由表）
 											</td>
 
 										</tr>
 										<tr>
-											<th>zerotier官网</th>
+											<th>Zerotier Central</th>
 											<td>
-				<input type="button" class="btn btn-success" value="zerotier官网" onclick="window.open('https://my.zerotier.com/network')" size="0">
-				<br>点击跳转到Zerotier官网管理平台，新建或者管理网络，并允许客户端接入访问你私人网路（新接入的节点默认不允许访问）
+				<input type="button" class="btn btn-success" value="Networks" onclick="window.open('https://my.zerotier.com/network')" size="0">
 											</td>
 										</tr>
 									</table>
 <table width="100%" align="center" cellpadding="4" cellspacing="0" class="table">
-	<tr> <th colspan="4">需要访问其它zerotier的内网LAN网段,IP和网关和zerotier后台对应即可(本机的LAN网段不用填进去)</th></tr>
+	<tr><th colspan="4" style="background-color: #E3E3E3;">Zerotier网络列表</th></tr>
                                         <tr id="row_rules_caption">
 										 
                                             <th width="10%">
                                                 启用 <i class="icon-circle-arrow-down"></i>
                                             </th>
 											<th width="20%">
-                                                IP <i class="icon-circle-arrow-down"></i>
+                                                网络ID <i class="icon-circle-arrow-down"></i>
                                             </th>
 											<th width="25%">
-                                                网关 <i class="icon-circle-arrow-down"></i>
+                                                备注 <i class="icon-circle-arrow-down"></i>
                                             </th>
                                             <th width="5%">
                                                 <center><i class="icon-th-list"></i></center>
@@ -275,10 +260,10 @@ function showMRULESList(){
 													<option value="0" <% nvram_match_x("","zero_enable_x_0", "0","selected"); %>>否</option>
 												</select>
                                            </th>
-                                         <th><input type="text" maxlength="255" class="span12" style="width: 200px" size="200" name="zero_ip_x_0" value="<% nvram_get_x("", "zero_ip_x_0"); %>"/>
+                                         <th><input type="text" maxlength="255" class="span12" style="width: 200px" size="200" name="zero_id_x_0" value="<% nvram_get_x("", "zero_id_x_0"); %>"/>
                                             </th>
                                          <th>
-                                                <input type="text" maxlength="255" class="span12" style="width: 200px" size="200" name="zero_route_x_0" value="<% nvram_get_x("", "zero_route_x_0"); %>" />
+                                                <input type="text" maxlength="255" class="span12" style="width: 200px" size="200" name="zero_note_x_0" value="<% nvram_get_x("", "zero_note_x_0"); %>" />
                                             </th>
                                          <th>
                                                 <button class="btn" style="max-width: 219px" type="submit" onclick="return markGroupRULES(this, 64, ' Add ');" name="markGroupRULES2" value="<#CTL_add#>" size="12"><i class="icon icon-plus"></i></button>
diff --git a/trunk/user/zerotier/zerotier.sh b/trunk/user/zerotier/zerotier.sh
index bc4aa82fa..7a3ccba80 100755
--- a/trunk/user/zerotier/zerotier.sh
+++ b/trunk/user/zerotier/zerotier.sh
@@ -31,7 +31,7 @@ start_instance() {
 		rm -f $config_path/identity.public
 	fi
 
-	add_join $(nvram get zerotier_id)
+	add_join
 
 	$PROG $args $config_path >/dev/null 2>&1 &
 
@@ -39,7 +39,14 @@ start_instance() {
 }
 
 add_join() {
-		touch $config_path/networks.d/$1.conf
+	netsnum=`nvram get zero_staticnum_x`
+	for i in $(seq 1 $rulesnum)
+	do
+		j=`expr $i - 1`
+		zero_enable=`nvram get zero_enable_x$j`
+		zero_id=`nvram get zero_id_x$j`
+		touch $config_path/networks.d/$zero_id.conf
+	done
 }
 
 
@@ -47,52 +54,37 @@ rules() {
 	while [ "$(ifconfig | grep zt | awk '{print $1}')" = "" ]; do
 		sleep 1
 	done
-	nat_enable=$(nvram get zerotier_nat)
-	zt0=$(ifconfig | grep zt | awk '{print $1}')
-	logger -t "zerotier" "zt interface $zt0 is started!"
 	del_rules
-	iptables -A INPUT -i $zt0 -j ACCEPT
-	iptables -A FORWARD -i $zt0 -o $zt0 -j ACCEPT
-	iptables -A FORWARD -i $zt0 -j ACCEPT
-	if [ $nat_enable -eq 1 ]; then
-		iptables -t nat -A POSTROUTING -o $zt0 -j MASQUERADE
-		while [ "$(ip route | grep "dev $zt0  proto" | awk '{print $1}')" = "" ]; do
-		sleep 1
-	    done
-		ip_segment=`ip route | grep "dev $zt0  proto" | awk '{print $1}'`
-		iptables -t nat -A POSTROUTING -s $ip_segment -j MASQUERADE
-		zero_route "add"
-	fi
-
+	zts=$(ifconfig | grep zt | awk '{print $1}')
+	nat_enable=$(nvram get zerotier_nat)
+	for ztn in $zts
+	do
+		logger -t "zerotier" "zt interface $ztn started!"
+		iptables -A INPUT -i $ztn -j ACCEPT
+		iptables -A FORWARD -i $ztn -o $ztn -j ACCEPT
+		iptables -A FORWARD -i $ztn -j ACCEPT
+		if [ $nat_enable -eq 1 ]; then
+			iptables -t nat -A POSTROUTING -o $ztn -j MASQUERADE
+			while [ "$(ip route | grep "dev $ztn  proto" | awk '{print $1}')" = "" ]; do
+				sleep 1
+			done
+			ip_segment=`ip route | grep "dev $ztn  proto" | awk '{print $1}'`
+			iptables -t nat -A POSTROUTING -s $ip_segment -j MASQUERADE
+		fi
+	done
 }
 
 del_rules() {
-	zt0=$(ifconfig | grep zt | awk '{print $1}')
-	ip_segment=`ip route | grep "dev $zt0  proto" | awk '{print $1}'`
-	iptables -D FORWARD -i $zt0 -j ACCEPT 2>/dev/null
-	iptables -D FORWARD -o $zt0 -j ACCEPT 2>/dev/null
-	iptables -D FORWARD -i $zt0 -o $zt0 -j ACCEPT
-	iptables -D INPUT -i $zt0 -j ACCEPT 2>/dev/null
-	iptables -t nat -D POSTROUTING -o $zt0 -j MASQUERADE 2>/dev/null
-	iptables -t nat -D POSTROUTING -s $ip_segment -j MASQUERADE 2>/dev/null
-}
-
-zero_route(){
-	rulesnum=`nvram get zero_staticnum_x`
-	for i in $(seq 1 $rulesnum)
+	zts=$(ifconfig | grep zt | awk '{print $1}')
+	for ztn in $zts
 	do
-		j=`expr $i - 1`
-		route_enable=`nvram get zero_enable_x$j`
-		zero_ip=`nvram get zero_ip_x$j`
-		zero_route=`nvram get zero_route_x$j`
-		if [ "$1" = "add" ]; then
-		if [ $route_enable -ne 0 ]; then
-			ip route add $zero_ip via $zero_route dev $zt0
-			echo "$zt0"
-		fi
-	else
-		ip route del $zero_ip via $zero_route dev $zt0
-	fi
+		ip_segment=`ip route | grep "dev $ztn  proto" | awk '{print $1}'`
+		iptables -D FORWARD -i $ztn -j ACCEPT 2>/dev/null
+		iptables -D FORWARD -o $ztn -j ACCEPT 2>/dev/null
+		iptables -D FORWARD -i $ztn -o $ztn -j ACCEPT
+		iptables -D INPUT -i $ztn -j ACCEPT 2>/dev/null
+		iptables -t nat -D POSTROUTING -o $ztn -j MASQUERADE 2>/dev/null
+		iptables -t nat -D POSTROUTING -s $ip_segment -j MASQUERADE 2>/dev/null
 	done
 }
 
@@ -108,7 +100,6 @@ kill_z() {
 stop_zero() {
 	logger -t "zerotier" "关闭zerotier"
 	del_rules
-	zero_route "del"
 	kill_z
 	rm -rf $config_path
 }
